practice/rain.py

```diff
@@ -0,0 +1,130 @@
+"""Takes data from rain gauge at Sunnyside and prints a summary of the data: date with the most rain &
+year with most rain.
+"""
+# setup:
+filename = 'sunnyside.txt'
+
+
+def each_line_of_file(filename):
+    """Opens sunnyside.txt and reads it line by line."""
+    with open(filename, newline='') as f:
+        lines = f.readlines()
+
+    return lines
+
+
+def split_lines(lines):
+    r"""Strips and splits the lines into lists.
+
+    >>> split_lines(['1 \n', '2\n', '3\n', '4\n', '5\n', '6 \n', '7 \n', '8 \n', '9 \n', '10 \n', '11 \n', 'PARTY! \n', 'TIME!'])
+    [['PARTY!']]
+    """
+    listed_lines = [x.strip().split('\n') for x in lines]
+    listed_lines_reduced = listed_lines[11:-1]
+    line_list = []
+    for listed_lines_split in listed_lines_reduced:
+        line_list += [y.split() for y in listed_lines_split]
+
+    return line_list
+
+
+def remove_junk(line_list):
+    """Cleans up the list and sets daily rain amount to an int.
+
+    >>> remove_junk([['test', '1', '2', '3']])
+    [['test', 1]]
+    """
+    temp_list = []
+    for z in line_list:
+        temp_list += [z[0:2]]
+    temp_list2 = []
+    for date, rain in temp_list:
+        if rain != '-':
+            int_rain = int(rain)
+            temp_list2 += [[date, int_rain]]
+    return temp_list2
+
+
+def dict_dates(y):
+    """Creates a dict from the list.
+
+    >>> dict_dates([['test', 1]])
+    {'test': 1}
+    """
+    d = {key: value for (key, value) in y}
+    d = {str(k): int(v) for k, v in d.items()}
+    return d
+
+
+def most_rain(d):
+    """Returns the key for max of dict daily totals.
+
+    >>> most_rain({'test': 1})
+    'test'
+    """
+    day_with_most_rain = max(d, key=d.get)
+    return day_with_most_rain
+
+
+def create_year_list(temp_list2):
+    """Slices the list of dates into years
+
+    >>> create_year_list([['Dec1999', 'Party over! OOPS! out of time']])
+    [['1999', 'Party over! OOPS! out of time']]
+    """
+    temp_list3 = []
+    for date, rain in temp_list2:
+        date_sliced = date[-4:]
+        temp_list3 += [[date_sliced, rain]]
+    return temp_list3
+
+
+def create_year_dict(temp_list3):
+    """Creates a dict from year list and sums rain value.
+
+    >>> create_year_dict([['1999', 1], ['1999', 2]])
+    {'1999': [3]}
+    """
+    yearly_totals = {}
+    for date_sliced, rain in temp_list3:
+        if date_sliced not in yearly_totals:
+            yearly_totals[date_sliced] = []
+        yearly_totals[date_sliced] += [rain]
+        total_by_year = dict(zip(yearly_totals.keys(), [[sum(item)] for item in yearly_totals.values()]))
```

Comment:
This re-calulation of yearly totals happens once after each new measurement is obtained. It only needs to happen after the whole grouping process is over once. Un-indent one level.